scripts/src/java/org/oppia/android/scripts/xml/TextViewStyleCheck.kt

```diff
@@ -0,0 +1,289 @@
+
+  val resDir = File(repoRoot, "app/src/main/res")
+  require(resDir.exists()) { "Resource directory does not exist: ${resDir.path}" }
+
+  val xmlFiles = resDir.listFiles { file -> file.isDirectory && file.name.startsWith("layout") }
+    ?.flatMap { dir -> dir.walkTopDown().filter { it.extension == "xml" } }
+    ?: emptyList()
+
+  val styleChecker = TextViewStyleCheck()
+  styleChecker.checkFiles(xmlFiles)
+}
+
+private class TextViewStyleCheck {
+  private val styleValidationIssues = mutableListOf<String>()
+  private val directionalityWarnings = mutableListOf<String>()
+  private val LINE_NUMBER_ATTRIBUTE = "lineNumber"
+
+  /** Checks XML files for TextView elements to ensure compliance with style requirements. */
+  fun checkFiles(xmlFiles: List<File>) {
+    xmlFiles.forEach { file -> processXmlFile(file) }
+    printResults()
+  }
+
+  private fun processXmlFile(file: File) {
+    val document = readXMLWithLineNumbers(FileInputStream(file), LINE_NUMBER_ATTRIBUTE)
+    val textViewNodes = document.getElementsByTagName("TextView")
+    val relativePath = file.path.substringAfter("main/res/")
+
+    for (i in 0 until textViewNodes.length) {
+      val element = textViewNodes.item(i) as Element
+      validateTextViewElement(element, relativePath)
+    }
+  }
+
+  private fun validateTextViewElement(element: Element, filePath: String) {
+    val lineNumber = element.getAttribute(LINE_NUMBER_ATTRIBUTE).toInt().minus(1).toString()
+    val styleAttribute = element.attributes.getNamedItem("style")?.nodeValue
+    val idAttribute = element.attributes.getNamedItem("android:id")?.nodeValue ?: "NO ID"
+
+    if (idAttribute in attributeIds) return
+
+    if (styleAttribute.isNullOrBlank()) {
+      styleValidationIssues.add(
+        "ERROR: Missing style attribute in file: $filePath, line $lineNumber."
+      )
+    }
+
+    checkForLegacyDirectionality(element, filePath, lineNumber)
+  }
+
+  private fun checkForLegacyDirectionality(
+    element: Element,
+    filePath: String,
+    lineNumber: String
+  ) {
+    val legacyAttributes = listOf(
+      "android:paddingLeft",
+      "android:paddingRight",
+      "android:layout_marginLeft",
+      "android:layout_marginRight",
+      "android:layout_alignParentLeft",
+      "android:layout_alignParentRight",
+      "android:layout_toLeftOf",
+      "android:layout_toRightOf"
+    )
+
+    if (legacyAttributes.any { element.hasAttribute(it) }) {
+      directionalityWarnings.add(
+        "WARNING: Hardcoded left/right attribute in file: $filePath, line $lineNumber. " +
+          "Consider using start/end."
+      )
+    }
+  }
+
+  private fun printResults() {
+    directionalityWarnings.forEach { println(it) }
+
+    if (styleValidationIssues.isNotEmpty()) {
+      styleValidationIssues.forEach { println(it) }
+      throw Exception("TEXTVIEW STYLE CHECK FAILED")
+    } else if (directionalityWarnings.isEmpty()) {
+      println("TEXTVIEW STYLE CHECK PASSED")
+    }
```

Comment:
```suggestion
      throw Exception("TEXTVIEW STYLE CHECK FAILED")
    } else {
      println("TEXTVIEW STYLE CHECK PASSED")
    }
```

We may not need to check for an empty `directionalityWarnings`, as it would prevent the 'PASSED' message from being printed if any warnings are present.